File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        for i, char_group in enumerate(zip(*strs)):
        # Check if all characters in the current group are the same
            if len(set(char_group)) == 1:
                continue
            else:
            # Found a mismatch, return the common prefix
                return strs[0][:i]

    # If all characters match up to the length of the shortest string, return that string
        return min(strs)
                    
            
            
            
            
                
        
        

File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String first=strs[0];
        String last=strs[strs.length - 1];
        int minLength = Math.min(first.length(), last.length());
        int i=0;
        while(i < minLength){
            if(first.charAt(i) == last.charAt(i)){
            i++;
            }
            else{
                break;
            }
        }
        return first.substring(0, i);
    }
}